mozilla_django_oidc/contrib/drf.py

```diff
@@ -0,0 +1,110 @@
+"""
+Classes/functions for integrating with Django REST Framework.
+
+http://www.django-rest-framework.org/api-guide/authentication/#custom-authentication
+"""
+
+from django.contrib.auth import get_backends
+from django.core.exceptions import ImproperlyConfigured, SuspiciousOperation
+from rest_framework import authentication, exceptions
+from requests.exceptions import HTTPError
+
+from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+from mozilla_django_oidc.utils import parse_www_authenticate_header
+
+
+def guess_oidc_backend():
+    """
+    Look through AUTH_BACKENDS to find one that is or extends the mozilla
+    OIDC backend class.
+    """
+    backends = [
+        backend for backend in get_backends()
+        if isinstance(backend, OIDCAuthenticationBackend)
+    ]
+    if not backends:
+        raise ImproperlyConfigured('No backends extending OIDCAuthenticationBackend found!')
+    if len(backends) > 1:
+        raise ImproperlyConfigured('More than one OIDCAuthenticationBackend found!')
+    return backends[0]
+
+
+class OIDCAuthentication(authentication.BaseAuthentication):
+    """
+    Provide OpenID authentication for DRF.
+    """
+
+    # used by the authenticate_header method.
+    www_authenticate_realm = 'api'
+
+    def __init__(self, backend=None):
+        self.backend = backend or guess_oidc_backend()
```

Comment:
Is there any case that we don't have OIDC enabled for the django app itself but only for the drf component? That way we imply that we use the authentication backend. Let's add a configuration entry to manually configure backend but also try to autoconfigure.

Reply:
Good point, I've added a specific setting for this.